src/utils/get-sorted-nodes.ts

```diff
@@ -54,8 +54,12 @@ export const getSortedNodes: GetSortedNodes = (nodes, options) => {
 
     for (const group of importOrder) {
         if (group === SEPARATOR_SPECIAL_WORD) {
-            if (!importOrderSeparation)
+            if (
+                finalNodes.length !== 0 &&
+                finalNodes[finalNodes.length - 1] !== newLineNode
+            ) {
```

Comment:
the `importOrderSeparation` no longer needs to be checked, since that is an implicit invariant based on the `finalNodes` - when `importOrderSeparation` is set, `finalNodes[finalNodes.length - 1]` will always be `newLineNode` https://github.com/joutvhu/prettier-plugin-sort-imports/pull/2/files#diff-8c9e4828f539d4af940aafe46d47c8d0251cc3692890eecfe0ea06d43a49ce65R83-R86